src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,45 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random;
+
+    public FigureSupplier(Random random) {
+        this.random = random;
+    }
+
+    public Figure getRandomFigure() {
+        ColorSupplier colorSupplier = new ColorSupplier(this.random);
+        int figureOption = random.nextInt(5);
+        Figure figure = new Figure();
+        switch (figureOption) {
+            case 0:
+                figure = new Square(colorSupplier.getRandomColor(), random.nextDouble());
+                break;
+            case 1:
+                figure = new Rectangle(colorSupplier.getRandomColor(), random.nextDouble(),
+                        random.nextDouble());
+                break;
+            case 2:
+                figure = new RightTriangle(colorSupplier.getRandomColor(), random.nextDouble(),
+                        random.nextDouble());
+                break;
+            case 3:
+                figure = new Circle(colorSupplier.getRandomColor(), random.nextDouble());
+                break;
+            case 4:
+                figure = new IsoscelesTrapezoid(colorSupplier.getRandomColor(), random.nextDouble(),
+                        random.nextDouble(), random.nextDouble());
+                break;
+            default:
+                break;
+        }
+
+        return figure;
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle("WHITE", 10);

```

Comment:
Magic number 10 and String of color you can take from Enum

Reply:
Why can't we use a number here?
![image](https://github.com/mate-academy/jv-oop-advanced/assets/71705897/c2744cdb-d910-4ca6-bbe7-3a4c431ada49)